src/main/java/net/snowflake/client/core/SFSession.java

```diff
@@ -1072,4 +1072,9 @@ private DriverPropertyInfo addNewDriverProperty(String name, String description)
     info.description = description;
     return info;
   }
+
+  /** @return whether this session uses async queries */
+  public boolean isAsyncSession() {
+    return !activeAsyncQueries.isEmpty();
```

Comment:
In this PR we only considered concurrent queries if the query is submitted explicitly via the asyncExecute API.

I'm not sure if it's possible (or maybe in other client drivers) that the same session can be shared by multiple application threads. If so, seems that we didn't handle those concurrent queries here. See https://docs.snowflake.com/en/sql-reference/transactions.html#transactions-and-multi-threading.

Any thoughts/comments on this? @sfc-gh-hchaturvedi @sfc-gh-vwakade @sfc-gh-emammedov